sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -61,5 +65,24 @@ public RequestFailedException(int status, string message, string? errorCode, Exc
             Status = status;
             ErrorCode = errorCode;
         }
+
+        /// <inheritdoc />
+        protected RequestFailedException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+            Status = info.GetInt32(nameof(Status));
+            ErrorCode = info.GetString(nameof(ErrorCode));
+        }
+
+        /// <inheritdoc />
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
```

Comment:
Do we need to implement both?

Reply:
Constructor is for reading and this method is for writing.

Reply:
FWIW, this is the standard serialization pattern (well, the old stuff, i.e. `[Serializable]` / `ISerializable`).